File: src/flutter/lib/services/counter/responses/entry_list_group_response.dart
import 'package:counter/services/counter/responses/entry_list_entry_response.dart';
import 'package:counter/services/counter/responses/entry_list_response.dart';

/// Represents a group in a [EntryListResponse].
class EntryListGroupResponse {

  final String name;
  final int total;
  final bool isEstimate;
  final List<EntryListEntryResponse> entries;
  const EntryListGroupResponse({
    required this.name,
    required this.total,
    required this.isEstimate,
    required this.entries
  });

  factory EntryListGroupResponse.fromJson(Map<String, dynamic> json) {
    return EntryListGroupResponse(
      name: json['name'] as String,
      total: json['total'] as int,
      isEstimate: json['isEstimate'] as bool,
      entries: List<EntryListEntryResponse>.from(
        json['entries'].map((entry) => EntryListEntryResponse.fromJson(entry)))
    );
  }
}

File: src/api/MintyPeterson.Counter.Api/Models/Responses/EntryListGroupResponse.cs
// <copyright file="EntryListGroupResponse.cs" company="Tom Cook">
// Copyright (c) Tom Cook. All rights reserved.
// </copyright>

namespace MintyPeterson.Counter.Api.Models.Responses
{
  /// <summary>
  /// Represents an group of <see cref="EntryListEntryResponse"/> objects in a <see cref="EntryListResponse"/>.
  /// </summary>
  public class EntryListGroupResponse
  {
    /// <summary>
    /// Gets or sets the group name.
    /// </summary>
    public string? Name { get; set; }

    /// <summary>
    /// Gets or sets the sum of <see cref="EntryListEntryResponse.Entry"/> in <see cref="Entries"/>.
    /// </summary>
    public decimal Total { get; set; }

    /// <summary>
    /// Gets or sets a value indicating whether the total is an estimate or not.
    /// </summary>
    public bool IsEstimate { get; set; }

    /// <summary>
    /// Gets or sets the <see cref="IEnumerable{EntryListEntryResponse}"/>.
    /// </summary>
    public IEnumerable<EntryListEntryResponse>? Entries { get; set; }
  }
}
